client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -1479,7 +1453,49 @@ public CompletableFuture<InsertResponse> insert(String tableName, InputStream da
      * @return {@code CompletableFuture<InsertResponse>} - a promise to insert response
      */
     public CompletableFuture<InsertResponse> insert(String tableName,
-                                     InputStream data,
+                                                    InputStream data,
+                                                    ClickHouseFormat format,
+                                                    InsertSettings settings) {
+
+        final int writeBufferSize = settings.getInputStreamCopyBufferSize() <= 0 ?
+                Integer.parseInt(configuration.getOrDefault(ClientConfigProperties.CLIENT_NETWORK_BUFFER_SIZE.getKey(),
+                        ClientConfigProperties.CLIENT_NETWORK_BUFFER_SIZE.getDefaultValue())) :
+                settings.getInputStreamCopyBufferSize();
+
+        if (writeBufferSize <= 0) {
+            throw new IllegalArgumentException("Buffer size must be greater than 0");
```

Comment:
Why we can not do that in `InsertSettings` and throw the exception from there?

Reply:
this value can be set by two method `setInputStreamCopyBufferSize` and `setOption` and here I'm doing specific calculation. 
One more reason - I keep validation here because it is implementation specific code for this operation. InsertSettings should not be aware of this validation because they may be used for other purposes.